xayn-ai/Cargo.toml

```diff
@@ -8,11 +8,12 @@ edition = "2018"
 anyhow = "1.0.52"
 bincode = "1.3.3"
 derivative = "2.2.0"
-derive_more = { version = "0.99.17", default-features = false, features = ["from", "display"] }
+derive_more = { version = "0.99.17", default-features = false, features = ["deref", "display", "from"] }
 displaydoc = "0.2.3"
 itertools = "0.10.3"
 layer = { path = "../layer" }
 lazy_static = "1.4.0"
+mockall_double = "0.2.1"
```

Comment:
I tried this in the past but it was impossible to have test with real implementation and mock during tests.